### PR TITLE
Add GitHub workflow using the Docker-based generator

### PR DESCRIPTION
**Pull request by rbehjati**

### CODE REVIEW

**Comment by rbehjati** [comment]
I was hoping that this would run the workflow. Not sure why it does not. Do you have any ideas? @mariaschett @tiziano88

**Comment by mariaschett** [comment]
I am not sure, but could it be that the workflow only runs after this PR is merged, i.e., with the next push?

File: .github/workflows/provenance.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  generate_provenance:
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_docker-based_slsa3.yml@main
+    with:
+      builder-image: "europe-west2-docker.pkg.dev/oak-ci/hello-transparent-release/hello-transparent-release"
+      builder-digest: "sha256:eb0297df0a4df8621837369006421dd972cc3e68e6da94625539f669d49f1525"
```
Comment (mariaschett) [inline .github/workflows/provenance.yaml:18]: Nice, I think this connects with the description in https://github.com/project-oak/hello-transparent-release#build-a-builder-docker-image .

Do you plan to update the README, too?

**Review by mariaschett — APPROVED** [review]
I think I'd try to merge.

File: .github/workflows/provenance.yaml (hunk 1 above)
Comment (rbehjati) [reply .github/workflows/provenance.yaml:18]: Yes. Once this is working I'll update the README too :)

**Comment by rbehjati** [comment]
> I am not sure, but could it be that the workflow only runs after this PR is merged, i.e., with the next push?

Might be, I'll merge and see if it helps.

**Comment by rbehjati** [comment]
Merge did trigger the workflow run :)

**Pull request merged**